<?xml version="1.0"?>
<!-- Launch file for the rrt-detector and the assigner -->
<launch>
    <arg name="eta"                     value="0.5"/>
    <arg name="Geta"                    value="2.0"/>
    <arg name="map_topic"               default="map"/>
    <arg name="odom_topic"              default="odom"/>
    <arg name="robot_base"              default="base_footprint"/>
    <arg name="global_frame"            default="map"/>
    <arg name="namespace"               default="/jetauto_"/>
    <arg name="n_robots"                default="1"/>
    <arg name="robot_name"              value="$(arg namespace)$(arg n_robots)"/>
    <param name="namespace_init_count"  value="1"/>

    <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
        <param name="eta"               value="$(arg Geta)"/>
        <param name="map_topic"         value="$(arg map_topic)"/>
        <param name="clicked_point"     value="/$(arg robot_name)/clicked_point"/>
        <param name="detected_points"   value="detected_points"/>
        <param name="shapes"            value="/$(arg robot_name)/global_detector_shapes"/>
    </node>

    <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
        <param name="eta"               value="$(arg eta)"/>
        <param name="map_topic"         value="$(arg map_topic)"/>
        <param name="robot_frame"       value="$(arg robot_base)"/>
        <param name="clicked_point"     value="/$(arg robot_name)/clicked_point"/>
        <param name="detected_points"   value="detected_points"/>
        <param name="shapes"            value="/$(arg robot_name)/local_detector_shapes"/>
    </node>

    <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
        <param name="map_topic"                     value="$(arg map_topic)"/>
        <param name="robot_frame"                   value="base_footprint"/>
        <param name="info_radius"                   value="0.8"/>
        <param name="costmap_clearing_threshold"    value="70"/>
        <param name="goals_topic"                   value="/$(arg robot_name)/detected_points"/>
        <param name="namespace"                     value="/$(arg namespace)"/>
        <param name="n_robots"                      value="$(arg n_robots)"/>
        <param name="rate"                          value="100"/>
        <param name="global_costmap_topic"          value="/move_base/global_costmap/costmap"/>
    </node>

    <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
        <param name="map_topic"                 value="$(arg map_topic)"/>
        <param name="global_frame"              value="$(arg global_frame)"/>
        <param name="robot_frame"               value="base_footprint"/>
        <param name="info_radius"               value="1"/>
        <param name="info_multiplier"           value="3.0"/>
        <param name="hysteresis_radius"         value="3.0"/>
        <param name="hysteresis_gain"           value="2.0"/>
        <param name="frontiers_topic"           value="/$(arg robot_name)/filtered_points"/>
        <param name="n_robots"                  value="$(arg n_robots)"/>
        <param name="namespace"                 value="/$(arg namespace)"/>
        <param name="delay_after_assignement"   value="0.5"/>
        <param name="rate"                      value="100"/>
        <param name="plan_service"              value="/move_base/GlobalPlanner/make_plan"/>
    </node>

    <node pkg="jetauto_slam" type="rrt_map_save.py" name="map_save" output="screen">
        <param name="map_frame"         value="$(arg global_frame)"/>
        <param name="odom_topic"        value="$(arg odom_topic)"/>
        <param name="clicked_point"     value="/$(arg robot_name)/clicked_point"/>
        <param name="wait_finish_time"  value="5"/>
    </node>
</launch>
